lightyear/src/client/prediction/rollback.rs

```diff
@@ -568,6 +569,38 @@ pub(crate) fn prepare_rollback_resource<R: Resource + Clone>(
     history.clear();
 }
 
+/// Return a fixed time that represents rollbacking `current_fixed_time` by
+/// `num_rollback_ticks` ticks. The returned fixed time's overstep is zero.
+///
+/// This function assumes that `current_fixed_time`'s timestep remained the
+/// same for the past `num_rollback_ticks` ticks.
+fn rollback_fixed_time(current_fixed_time: &Time<Fixed>, num_rollback_ticks: i16) -> Time<Fixed> {
+    let mut rollback_fixed_time = Time::<Fixed>::from_duration(current_fixed_time.timestep());
+    if num_rollback_ticks <= 0 {
+        debug!("Cannot rollback fixed time by {} ticks", num_rollback_ticks);
+        return rollback_fixed_time;
+    }
+    // Fixed time's elapsed time's is set to the fixed time's delta before any
+    // fixed system has run in an app, see
+    // `bevy_time::fixed::run_fixed_main_schedule()`. If elapsed time is zero
+    // that means no tick has run.
+    if current_fixed_time.elapsed() < current_fixed_time.timestep() {
+        error!("Current elapsed fixed time is less than the fixed timestep");
+        return rollback_fixed_time;
+    }
+
+    // Difference between the current time and the time of the first tick of
+    // the rollback.
+    let rollback_time_offset = (num_rollback_ticks - 1) as u32 * rollback_fixed_time.timestep();
+
+    let rollback_elapsed_time = current_fixed_time.elapsed() - rollback_time_offset;
+    rollback_fixed_time.advance_to(rollback_elapsed_time - rollback_fixed_time.timestep());
+    // This sets the fixed time's delta to the correct amount.
+    rollback_fixed_time.advance_by(rollback_fixed_time.timestep());
```

Comment:
I think this could have a clarifying comment.
`Time<Fixed>.delta` is set to the value provided in `advance_by` (or `advance_to`), so we want to call `advance_by(rollback_fixed_time.timestep())` at the end to set the delta value that is expected.

(I had to read the `advance_by` source to check this, but it would be good to mention it here)

Reply:
Updated!